programs/dynamic-bonding-curve/src/instructions/creator/ix_transfer_pool_creator.rs

```diff
@@ -0,0 +1,46 @@
+use anchor_lang::prelude::*;
+
+use crate::{
+    state::{MigrationProgress, VirtualPool},
+    EvtUpdatePoolCreator, PoolError,
+};
+
+/// Accounts for transfer pool creator
+#[event_cpi]
+#[derive(Accounts)]
+pub struct TransferPoolCreatorCtx<'info> {
+    #[account(
+        mut,
+        has_one = creator,
+    )]
+    pub virtual_pool: AccountLoader<'info, VirtualPool>,
+
+    pub creator: Signer<'info>,
+
+    /// CHECK: new creator address, can be anything except old creator
+    #[account(
+        constraint = new_creator.key().ne(creator.key) @ PoolError::InvalidNewCreator,
+    )]
+    pub new_creator: UncheckedAccount<'info>,
+}
+
+pub fn handle_transfer_pool_creator(ctx: Context<TransferPoolCreatorCtx>) -> Result<()> {
+    let mut pool = ctx.accounts.virtual_pool.load_mut()?;
+
+    let migration_progress = pool.get_migration_progress()?;
+    // avoid pool creator to do update between 2 periods
+    require!(
+        migration_progress == MigrationProgress::PreBondingCurve
+            || migration_progress == MigrationProgress::CreatedPool,
```

Comment:
What will happen if
Graduate to: DAMM v1
Transaction sequence:
1. Migrate DAMM v1 pool
2. Lock LP (Trade fees belong to old creator)
3. Integrator transfer pool creator
4. Claim LP (Trade fees belong to new creator)

or vice versa.

This causes more handling required for integrator program. Maybe we can add to avoid "partial state" by checking `locked_lp && claimed_lp || !locked_lp && !claimed_lp`?

Note: Aware of current operation is bundled. But just in case?

Reply:
On damm v2, no issue
On damm v1, they can transfer LP token if they already claimed it, while we haven't supported to transfer locked escrow, so they still need to claim for creator